### ISSUE TITLE
Switching ICU Message Format for something else?

### ISSUE THREAD

**Issue by vvohra** [yahoo/react-intl#672]
Just wanted to say Format.js and related projects are really great and I'd like to integrate it into some projects.

The biggest issue right now is that my translation workflow is tied heavily to the current Translation Management Service (TMS) using gettext/po files. Unfortunately the current TMS doesn't have any real support for ICU Message Format in their tools (mostly verification while vendors are translating, great tooling around capturing screenshots/HTML for translation context). 

I've read a few GitHub issues where others are kinda 'stuck' like I am as well. ICU is really great, but the TMS support and tooling around it isn't quite there. And building a TMS alternative with full ICU support is out of scope for me at this moment.

So I'm kinda stuck, there are really great parts of Ember/React-Intl (number/currency/relative formatting) which I can get through mix/matching some libs or rolling out my own but the message formatting is holding me back from adopting this project.

But in a perfect world, what would be great is that I could 'switch' out the ICU Message Format with an alternative message format. I'm spending some time to see if I can do this, but I wanted to reach out to you if you had any thoughts.

**Comment by ericf** [comment]
What's the alternative format? And what features are different from ICU?

You can always use [`intl-messageformat-parser`](https://github.com/yahoo/intl-messageformat-parser) to parse the ICU messages then generate a new output format, but you'd also have to do the reverse parse/generation.

**Comment by vvohra** [comment]
So in gettext/po files for english I would define: "{{count}} message"/"{{count}} messages" for the singular/plural forms. And the plural forms will be defined for the various locales. Feels like a very traditional gettext/po format.

I agree it isn't as capable as ICU Message Format with regards to gender and some other things.

Parsing the ICU to the output format is certainly possible, that's interesting. I'd have to think some cases where that might get a little complicated.

**Comment by vvohra** [comment]
> You can always use intl-messageformat-parser to parse the ICU messages then generate a new output format, but you'd also have to do the reverse parse/generation.

After thinking about it, I think I can actually accomplish the conversion in both ways if I only support a simpler version of the ICU. Thanks for the suggestion, I'll see if I can come up with something.

**Comment by vvohra** [comment]
For anyone who might look at this in the future.

What I'm doing is supporting a simplified ICU Message Format as our tooling gets better. For example:

`viewers_label: {numViewers, plural, one {1 viewer} other {{numViewers} viewers}}`

becomes:

```
# plural: numViewers
viewers_label: 
   one: 1 viewer
   other: {numViewers} viewers
```
for translation purposes. 

Once the translation is done, I merge this back to the original format.

This means:
- Don't support complex ICU Messages like: multiple plurals
- Pretty much only support the plural, do not support: select, gender
- Still support the formatting arguments because using {...} to detect placeholders is valid.

This works for me because I'm using this as a replacement of GetText/PO and I've gotten feature parity. And as we improve our tools, we'll be able to use more features of ICU Message Format.

**Comment by slorber** [comment]
Here's what I found:

https://www.npmjs.com/package/format-message-print
http://format-message.github.io/icu-message-format-for-translators/editor.html
